### PR TITLE
Only visual studio 2015 neesd fixes the windows sdk path

### PR DESCRIPTION
**Pull request by lygstate**
### This changes visible behavior

The following changes are proposed:
Currently, the fixes windows sdk also works for vs2022, but that not needed, narrow down to vs2015 only

### CODE REVIEW

**Comment by gcampbell-msft** [comment]
@lygstate Could you provide more context on why this change is needed and what scenarios it fixes? Thanks

**Comment by lygstate** [comment]
> @lygstate Could you provide more context on why this change is needed and what scenarios it fixes? Thanks

When I first introudce this fix, did not notice that it's also fixing for vs2022 env bat,
That is
```
[visual-studio] Patch Windows SDK path from C:\Program Files (x86)\Windows Kits\10\bin\x64 to C:\Program Files (x86)\Windows Kits\10\bin\10.0.26100.0\x64 for C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat
```

Only ` C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat` is invalid, but for vs2022, there is no such issue, so there is no need patch it

File: src/installs/visualStudio.ts [hunk 1]
```diff
@@ -407,7 +407,7 @@ async function collectDevBatVars(hostArch: string, devBat: string, args: string[
     } catch (err) {
         log.error(`Parse '${WindowsSDKVersion}' failed`);
     }
-    if (util.compareVersion(WindowsSDKVersionParsed, { major: 10, minor: 0, patch: 14393 }) >= 0) {
+    if (majorVersion === 14 && util.compareVersion(WindowsSDKVersionParsed, { major: 10, minor: 0, patch: 14393 }) >= 0) {
         const WindowsSdkDir = vars['WindowsSdkDir'] ?? '';
```
[the remark this answers is not in the document]
Comment (gcampbell-msft) [reply src/installs/visualStudio.ts:411]: Does it cause issues without this fix?

[the remark this answers is not in the document]
Comment (lygstate) [reply src/installs/visualStudio.ts:411]: It's annoy report misleading extention message that fix windows sdk for vs2022

[the remark this answers is not in the document]
Comment (gcampbell-msft) [reply src/installs/visualStudio.ts:411]: Ahhh, I understand more now, I'll reopen

**Comment by gcampbell-msft** [comment]
@lygstate I'll approve but please make a CHangelog update.

**Review by gcampbell-msft — APPROVED** [review]

**Review by gcampbell-msft — APPROVED** [review]

**Review by gcampbell-msft — APPROVED** [review]

**Pull request merged**